spec/services/get_product_service_spec.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe GetProductService do
+  subject(:service) { described_class.new(product_id) }
+
+  let(:product_id) { SecureRandom.uuid }
+
+  let!(:category) { create(:category) }
+  let!(:sizes) { create_list(:size, 2, product: second_product) }
+  let!(:toppings) { create_list(:topping, 2, product: second_product) }
+  let!(:tag) { create(:tag, product: second_product) }
+  let!(:first_product) { create(:product, category:, name: 'Cà phê') }
+  let!(:second_product) { create(:product, id: product_id, category:, name: 'Trà sữa') }
+
+  describe '#validate' do
+    context 'when product_id is nil' do
```

Comment:
how about case when there is no product with given product_id exists in db ?

Reply:
Added that case, pls check again